### PR TITLE
CI for Mac

### LINKED ISSUE #722: CI for mac
To prevent things like #721

### PR DESCRIPTION
**Pull request by mrks**
I did not find a nice way to run the traditional FullCI steps and this in parallel. It is possible, but Blue Ocean cannot visualize it properly. In the end, all steps are flying around and it's hard to find something. Thus, it is its own phase right now. Other systems (e.g., Power, Raspberry, ...) can be added there without too much hassle (and those would run in parallel).

fixes #722

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -36,7 +33,7 @@ if echo $REPLY | grep -E '^[Yy]$' > /dev/null; then
                 fi
             done
 
-            if ! brew install llvm --with-toolchain; then
+            if ! brew install llvm; then
```
Comment (mrks) [inline install.sh]: Not necessary/supported anymore

File: src/test/lossless_cast_test.cpp [hunk 2]
```diff
@@ -113,9 +113,9 @@ TEST_F(LosslessCastTest, IntegralToIntegral) {
   EXPECT_EQ(lossless_cast<int32_t>(int64_t(3)), 3);
 
   EXPECT_EQ(lossless_cast<int32_t>(2'147'483'647), 2'147'483'647);
-  EXPECT_EQ(lossless_cast<int32_t>(int64_t{2'147'483'648}), std::nullopt);
-  EXPECT_EQ(lossless_cast<int32_t>(int64_t{-2'147'483'648}), -2'147'483'648);
-  EXPECT_EQ(lossless_cast<int32_t>(int64_t{-2'147'483'649}), std::nullopt);
+  EXPECT_EQ(lossless_cast<int32_t>(2'147'483'648), std::nullopt);
```
Comment (mrks) [inline src/test/lossless_cast_test.cpp]: Broke this on purpose to make the CI red

File: install.sh [hunk 3]
```diff
@@ -15,9 +15,6 @@ if echo $REPLY | grep -E '^[Yy]$' > /dev/null; then
             exit 1
         fi
 
-        # Needed for proper building under macOS
-        xcode-select --install
```
Comment (mrks) [inline install.sh]: We require xcode to be installed prior to executing this, but the command line tools come automatically

File: Jenkinsfile [hunk 4]
```diff
@@ -220,31 +223,51 @@ node {
...
+node('mac') {
+  stage("clangDebugMac") {
+    if (env.BRANCH_NAME == 'master' || full_ci) {
+      try {
+        checkout scm
+
+        // We do not use install.sh here as there is no way to run OS X in a Docker container
+        sh "git submodule update --init --recursive --jobs 4"
+
+        sh "mkdir clang-debug && cd clang-debug && /usr/local/bin/cmake -DCMAKE_CXX_COMPILER_LAUNCHER=/usr/local/bin/ccache -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=/usr/local/Cellar/llvm\\@7/7.0.1/bin/clang -DCMAKE_CXX_COMPILER=/usr/local/Cellar/llvm\\@7/7.0.1/bin/clang++ .."
```
Comment (mrks) [inline Jenkinsfile]: Forced to llvm7 as 6 does not have std::filesystem on Mac. With #1433, this will be the default anyway.

File: Jenkinsfile [hunk 5]
```diff
@@ -173,7 +176,7 @@ node {
             sh "TSAN_OPTIONS=suppressions=resources/.tsan-ignore.txt ./clang-release-thread-sanitizer/hyriseTest clang-release-thread-sanitizer"
             sh "TSAN_OPTIONS=suppressions=resources/.tsan-ignore.txt ./clang-release-thread-sanitizer/hyriseSystemTest clang-release-thread-sanitizer"
           } else {
-            Utils.markStageSkippedForConditional("clangReleaseThreadSanitizerNoNuma")
+            Utils.markStageSkippedForConditional("clangReleaseThreadSanitizer")
```
Comment (mrks) [inline Jenkinsfile]: Missed that in #1618

File: Jenkinsfile (hunk 4 above)
Comment (Bensk1) [inline Jenkinsfile]: Just for clarification what is the `\\@7` necessary for?

File: Jenkinsfile [hunk 6]
```diff
@@ -220,31 +223,51 @@ node {
...
+      try {
+        checkout scm
+
+        // We do not use install.sh here as there is no way to run OS X in a Docker container
+        sh "git submodule update --init --recursive --jobs 4"
+
+        sh "mkdir clang-debug && cd clang-debug && /usr/local/bin/cmake -DCMAKE_CXX_COMPILER_LAUNCHER=/usr/local/bin/ccache -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=/usr/local/Cellar/llvm\\@7/7.0.1/bin/clang -DCMAKE_CXX_COMPILER=/usr/local/Cellar/llvm\\@7/7.0.1/bin/clang++ .."
+        sh "cd clang-debug && CCACHE_CPP2=yes CCACHE_SLOPPINESS=file_macro PATH=/usr/local/bin:$PATH make -j libjemalloc-build"
+        sh "cd clang-debug && CCACHE_CPP2=yes CCACHE_SLOPPINESS=file_macro make -j4"
+        sh "cd clang-debug && ./hyriseTest"
```
Comment (Bensk1) [inline Jenkinsfile]: Any specific reason for not using the typical `./clang-debug/hyriseTest`?

File: Jenkinsfile [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
 
-node {
+try {  // TODO fix indentation - change for easier reviewing
+node('master') {
```
[the remark this answers is not in the document]
Comment (mrks) [reply Jenkinsfile]: It's not a node, but a node label. "Run this on any node that is tagged 'master'".

File: Jenkinsfile (hunk 4 above)
Comment (mrks) [reply Jenkinsfile]: `brew install llvm` will give you clang-8. To install clang-7, we need `brew install llvm@7`, which creates `/usr/local/Cellar/llvm@7`. The `@` needs to be escaped for some reason, and as we are in a string, the escape character needs to be escaped, too.

File: Jenkinsfile (hunk 6 above)
Comment (mrks) [reply Jenkinsfile]: None in particular. Will change.

**Comment by mrks** [comment]
> I am not an expert in manually parsing Jenkinsfiles

And I'm not an expert in writing them. If it works and provably runs on the Mac Mini, it can't be too bad, I guess... ;)

File: Jenkinsfile (hunk 4 above)
Comment (Bensk1) [reply Jenkinsfile]: Understood.

**Pull request merged**